runtime/src/main/java/org/corfudb/runtime/view/CorfuGuidGenerator.java

```diff
@@ -244,12 +255,9 @@ private synchronized CorfuGuid getCorrectedTimestamp() {
                 }
                 resolutionCorrection = 0;
             }
-        } else {
-            resolutionCorrection = 0;
-        }
-
-        if (currentTimestamp < previousTimestamp) {
+        } else if (currentTimestamp < previousTimestamp) { // NTP corrects time backward
```

Comment:
why do you need the condition? we already checked >, and =...

Reply:
they are different right - currentTime > previousTimestamp means normal time moving forward, we only need to check for crazy jumps,
currentTime == previousTime means this call was invoked within the same millisecond resolution
But if currentTime < previousTime means time has moved backward and monotonic growth is broken so we fix up by adding drift correction.

Reply:
correct, but if it is not greater and it is not equal, it necessarily is lower, so an else would be enough, right? (without the if)

Reply:
oh, got it, sure, I can remove the redundant if